Source/diablo.cpp

```diff
@@ -933,15 +924,14 @@ void DiabloParseFlags(int argc, char **argv)
 		} else if (strcasecmp("--save-dir", argv[i]) == 0) {
 			paths::SetPrefPath(argv[++i]);
 		} else if (strcasecmp("--demo", argv[i]) == 0) {
-			demoMode = true;
-			if (!LoadDemoMessages(SDL_atoi(argv[++i]))) {
-				SDL_Log("Unable to load demo file");
-				diablo_quit(1);
-			}
+			demoNumber = SDL_atoi(argv[++i]);
+			gbShowIntro = false;
 		} else if (strcasecmp("--timedemo", argv[i]) == 0) {
 			timedemo = true;
 		} else if (strcasecmp("--record", argv[i]) == 0) {
-			recordDemo = SDL_atoi(argv[++i]);
+			recordNumber = SDL_atoi(argv[++i]);
+		} else if (strcasecmp("--create-reference", argv[i]) == 0) {
+			createDemoReference = true;

```

Comment:
I still think we should find a different an probably more automated way to achieve this, but unless I think of something by the end of this review then we can keep this for now.

Reply:
How about simply always doing this for debug builds?

Reply:
If you want I can do this. But I thought a additional parameter doesn't harm anyone.